ocs_ci/deployment/deployment.py

```diff
@@ -605,9 +606,21 @@ def deploy_ocs(self):
         # Verify health of ceph cluster
         # TODO: move destroy cluster logic to new CLI usage pattern?
         logger.info("Done creating rook resources, waiting for HEALTH_OK")
-        assert ceph_health_check(
-            namespace=self.namespace
-        )
+        try:
+            ceph_health_check(namespace=self.namespace, tries=30, delay=10)
+        except CephHealthException as ex:
+            err = str(ex)
+            logger.warning(f"Ceph health check failed with {err}")
+            if "clock skew detected" in err:
+                logger.info(
+                    f"Changing NTP on compute nodes to"
+                    f" {constants.RH_NTP_CLOCK}"
+                )
+                update_ntp_compute_nodes()
```

Comment:
Instead of manually applying ntp change we can make use of https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/templates/ocp-deployment/ntp_chrony.yaml rest of the things will be taken care by ocp

Reply:
+1

Reply:
If the proposed solution works the implementation should be much simpler @prsurve right?

Reply:
yes I have tried using YAML to fix clock skew and its works fine

Reply:
after applying we have to wait for some time (approx 10 min in total) we can check the progress from here https://github.com/red-hat-storage/ocs-ci/blob/b900584d1affd5cf37b906b6a51699eabef37b9f/ocs_ci/utility/utils.py#L2364-L2378

Reply:
Today I tried the yaml file and it looks like it restarts the the worker machines. It almost took 10 minutes to bring up the 3 workers. If we bring up the chrony at the end of OCS installation once skew is detected, then once worker machines is up after 10 minutes, we need to wait for another ~5 minutes, since some PG's are degraded after worker machines got rebooted.

Another option is we can bring up the chrony after OCP installation, but we have to accept the fact that we are rebooting all the workers even though we don't have skew issue.  In Later case I didn't try the OCS installation.

Reply:
yeah reboot looks like bad option.

Reply:
Weird is that on the same DC it sometime works and sometime not, even if we set the same NTP server on all machines I would expect that VMs will take the time from hypervisor.  I am just want to avoid to implement such functionality for all platforms if it will be needed at some point.  If we will follow this settings o NTP on all machines we will be sure we should have it set all the time and later on we will not see any discrepancy in the execution later on which can maybe happen.     When I went quickly over:  https://access.redhat.com/solutions/4906341 This is the right way how they describe this to be done.  From my understanding if we will do this in the OCP deployment then we will have the config from the start and no need for the restart at all.

The thing is how easy it will be for us to cover this in the deployment. Wanted to have quick sync mtg with you @vavuthu . So let me know when you have 10 mins.